From 5a406c361480cf600f30b6384bad931c0e845a91 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Wed, 26 Apr 2017 17:20:47 +0300
Subject: [PATCH 0013/1239] phy: marvell: add missing speed during info prints

In get_speed_string() we have an array (speed_strings[])
which includes all possible speed strings.
This array size and content must be aligned to the speed
defines in comphy_data.h.

This patch adds missing 5.125G speed, aligns speed_strings[]
and fixes incorrect printing when speed > 5.0G.

Change-Id: I9900d23595094be321be0c62fcaa88036324568e
Signed-off-by: Igal Liberman <igall@marvell.com>
---
 drivers/phy/marvell/comphy_core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/marvell/comphy_core.c b/drivers/phy/marvell/comphy_core.c
index 3065b87ebc..f3725312b1 100644
--- a/drivers/phy/marvell/comphy_core.c
+++ b/drivers/phy/marvell/comphy_core.c
@@ -21,8 +21,9 @@ static const char *get_speed_string(u32 speed)
 {
 	static const char * const speed_strings[] = {
 		"1.25 Gbps", "1.5 Gbps", "2.5 Gbps",
-		"3.0 Gbps", "3.125 Gbps", "5 Gbps", "6 Gbps",
-		"6.25 Gbps", "10.31 Gbps"
+		"3.0 Gbps", "3.125 Gbps", "5 Gbps",
+		"5.125 Gpbs", "6 Gbps", "6.25 Gbps",
+		"10.3125 Gbps"
 	};
 
 	if (speed < 0 || speed > COMPHY_SPEED_MAX)
-- 
2.29.0

